util/keypair.ts

```diff
@@ -0,0 +1,51 @@
+export async function generateKeyPair(): Promise<{
+    publicKey: CryptoKey,
+    privateKey: CryptoKey,
+    exportedPublicKey: ArrayBuffer,
+    exportedPrivateKey: ArrayBuffer,
+    publicKeyString: string,
+    privateKeyString: string,
+    fingerprint: string
+}> {
+    const keyPair = await crypto.subtle.generateKey(
+        {
+            name: 'RSASSA-PKCS1-v1_5',
```

Comment:
how'd you pick this? from a [quick read](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey#parameters), it looks like `RSASSA-PKCS1-v1_5` and `RSA-PSS` are both intended for signing, whereas `RSA-OEAP` is for encryption/decryption -- my guess is `RSA-OEAP` is what you want